### PR TITLE
[mnt][Part2] Optimize rbf & linear kernel for SOA table result 

### PR DESCRIPTION
**Pull request by PetrovKP**
Merge from https://github.com/oneapi-src/oneDAL/pull/921

### CODE REVIEW

File: algorithms/kernel/svm/svm_train_thunder_impl.i [hunk 1]
```diff
@@ -184,13 +184,12 @@ services::Status SVMTrainImpl<thunder, algorithmFPType, ParameterType, cpu>::com
...
 
-        DAAL_CHECK_STATUS(status, cachePtr->getRowsBlock(wsIndices, nWS, kernelWS));
-        DAAL_CHECK_STATUS(
-            status, SMOBlockSolver(y, grad, wsIndices, kernelWS, nVectors, nWS, cw, eps, tau, buffer.get(), I.get(), alpha, deltaAlpha.get(), diff));
-
-        DAAL_CHECK_STATUS(status, updateGrad(kernelWS, deltaAlpha.get(), gradBuff.get(), grad, nVectors, nWS));
+        DAAL_CHECK_STATUS(status, updateGrad(kernelSOARes, deltaAlpha.get(), gradBuff.get(), grad, nVectors, nWS));
 
         if (checkStopCondition(diff, diffPrev, eps, sameLocalDiff) && iter >= nNoChanges) break;
         diffPrev = diff;
```
Comment (owerbat) [inline algorithms/kernel/svm/svm_train_thunder_impl.i:195]: You probably missed `cachePtr->clear();` after this block as in #921

File: algorithms/kernel/svm/svm_train_thunder_impl.i [hunk 2]
```diff
@@ -222,19 +221,17 @@ services::Status SVMTrainImpl<thunder, algorithmFPType, ParameterType, cpu>::SMO
...
-                const size_t wsIndex = wsIndices[i];
-
-                ReadColumns<algorithmFPType, cpu> mtKernel(kernelWS.get(), i, 0, nVectors);
-                DAAL_CHECK_BLOCK_STATUS_THR(mtKernel);
-                const algorithmFPType * const kernelWSData = mtKernel.get();
+                const size_t wsIndex                       = wsIndices[i];
+                const algorithmFPType * const kernelWSData = kernelWS[i];
                 yLocal[i]                                  = y[wsIndex];
                 gradLocal[i]                               = grad[wsIndex];
                 oldAlphaLocal[i]                           = alpha[wsIndex];
```
Comment (owerbat) [inline algorithms/kernel/svm/svm_train_thunder_impl.i:237]: And you probably forgot `PRAGMA_IVDEP; PRAGMA_VECTOR_ALWAYS` before `for` at line 245

File: algorithms/kernel/svm/svm_train_thunder_impl.i (hunk 1 above)
Comment (PetrovKP) [reply algorithms/kernel/svm/svm_train_thunder_impl.i:195]: thanks a lot

File: algorithms/kernel/svm/svm_train_thunder_impl.i (hunk 2 above)
Comment (PetrovKP) [reply algorithms/kernel/svm/svm_train_thunder_impl.i:237]: it is not necessary. I will delete in the master.

**Comment by PetrovKP** [comment]
http://intel-ci.intel.com/eaf68503-7c1c-f1fa-b91a-8cdcd4b723a1

**Review by owerbat — APPROVED** [review]

**Pull request merged**